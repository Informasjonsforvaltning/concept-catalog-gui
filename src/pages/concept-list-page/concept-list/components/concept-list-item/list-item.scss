@import '../../../../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors';

.fdk-list-item {
  background-color: $color-white;
  color: $color-neutral-dark;
  margin-bottom: 0.25em;
  border-radius: 3px;
  padding: 0.5em 1em;
  text-decoration: none;
}
