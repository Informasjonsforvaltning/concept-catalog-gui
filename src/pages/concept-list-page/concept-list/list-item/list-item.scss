@import '../../../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-list-item {
  background-color: $color-white;
  color: $color-dark-1;
  margin-bottom: 0.25em;
  border-radius: 3px;
  padding: 0.5em 1em;
  text-decoration: none;
}
