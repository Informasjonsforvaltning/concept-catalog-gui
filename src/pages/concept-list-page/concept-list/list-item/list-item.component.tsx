import React from 'react';
import { Link } from 'react-router-dom';
import cx from 'classnames';

import { localization } from '../../../../lib/localization';
import './list-item.scss';

interface Props {
  col1: string;
  col2: string;
  col3: string;
  status: string;
  path: string;
}

export const ListItem = ({ col1, col2, col3, status, path }: Props): JSX.Element | null => {
  if (!(col1 || status)) {
    return null;
  }

  const statusClass = cx('fa', 'fa-circle', 'mr-2', 'd-none', 'd-sm-inline', {
    'fdk-color-cta': status === 'PUBLISH',
    'fdk-color3': status === 'DRAFT'
  });

  return (
    <Link to={path} className="row fdk-list-item">
      <span className="col-3">{col1}</span>
      <span className="col-3">{col2}</span>
      <span className="col-3">{col3}</span>
      <span className="col-3">
        <i className={statusClass} />
        {status === 'PUBLISH' && <span>{localization['published']}</span>}
        {status === 'DRAFT' && <span>{localization['draft']}</span>}
      </span>
    </Link>
  );
};
