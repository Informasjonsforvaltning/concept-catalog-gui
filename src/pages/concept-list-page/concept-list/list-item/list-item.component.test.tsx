import React from 'react';
import { shallow } from 'enzyme';
import { ListItem } from './list-item.component';

let defaultProps;

beforeEach(() => {
  defaultProps = {
    col1: 'column1',
    col2: 'column2',
    col3: 'column3',
    path: '/tester'
  };
});

test('should render ListItem with status DRAFT', (): void => {
  const wrapper = shallow(<ListItem {...defaultProps} status='DRAFT' />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ListItem with status PUBLISH', (): void => {
  const wrapper = shallow(<ListItem {...defaultProps} status='PUBLISH' />);
  expect(wrapper).toMatchSnapshot();
});
