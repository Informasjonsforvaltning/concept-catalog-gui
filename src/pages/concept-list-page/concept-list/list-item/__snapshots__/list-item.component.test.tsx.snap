// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ListItem with status DRAFT 1`] = `
<Link
  className="row fdk-list-item"
  to="/tester"
>
  <span
    className="col-3"
  >
    column1
  </span>
  <span
    className="col-3"
  >
    column2
  </span>
  <span
    className="col-3"
  >
    column3
  </span>
  <span
    className="col-3"
  >
    <i
      className="fa fa-circle mr-2 d-none d-sm-inline"
    />
  </span>
</Link>
`;

exports[`should render ListItem with status PUBLISH 1`] = `
<Link
  className="row fdk-list-item"
  to="/tester"
>
  <span
    className="col-3"
  >
    column1
  </span>
  <span
    className="col-3"
  >
    column2
  </span>
  <span
    className="col-3"
  >
    column3
  </span>
  <span
    className="col-3"
  >
    <i
      className="fa fa-circle mr-2 d-none d-sm-inline"
    />
  </span>
</Link>
`;
